common/reviews/api/terminal.api.md

```diff
@@ -57,12 +159,47 @@ export interface IStdioSummarizerOptions {
     trailingLines?: number;
 }
 
+// @beta (undocumented)
+export interface IStringBufferOutputOptions {
+    normalizeSpecialCharacters: boolean;
+}
+
+// @beta (undocumented)
+export interface ITerminal {
+    registerProvider(provider: ITerminalProvider): void;
+    unregisterProvider(provider: ITerminalProvider): void;
+    write(...messageParts: WriteParameters): void;
```

Comment:
```ts
export type WriteParameters = string[] | [...string[], IWriteOptions];
```


The name `WriteParameters` makes sense in the context of this function, but it is confusing as a top-level export especially alongside  `WritableOptions` from `stream` and also `IWriteOptions`.

I would suggest to rename it to `TerminalWriteParameters` for clarity.

HOWEVER we might also question the usefulness of this array.  Why is it valuable to write...
```ts
myTerminal.write('Hello', ' world!', {doNotOverrideSgrCodes: true});
```
...instead of this?
```ts
myTerminal.write('Hello' + ' world!', {doNotOverrideSgrCodes: true});
```
It is not saving much keystrokes, and it complicates the implementation of `write()` particularly for proxies who must pass these parameters on to other layers.  For a potentially very high bandwidth API it could also be a performance concern.

It seems plenty sufficient to define it like this:
```ts
write(message: string, options: ITerminalWriteOptions): void;
```

Reply:
Looking at the underlying implementation, the processing of an array does help to provide backwards compatibility (at runtime apparently, without typings) for the old colorable sequence structures.  However:

1. If we wanted to accomplish that goal, it would need to be an untyped export from `node-core-library` not `terminal`, right?  And the array processing baggage would belong there, not here.
2. How important is this backwards compatibility? For a MAJOR version bump, maybe it's okay to skip that requirement. (?)

Reply:
The value in keeping the legacy support here is when Heft/Rush plugins use the terminal from an updated version of Heft without being updated themselves, they'll continue to work.

Reply:
This will avoid breaking existing Heft and Rush plugins.

I propose we make this change in a follow-up where we remove the legacy support in Terminal and bump Heft to 1.0.